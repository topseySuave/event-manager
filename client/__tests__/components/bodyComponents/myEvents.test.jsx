import React from 'react';
import { shallow } from 'enzyme';
import { MyEvents } from
  '../../../src/components/bodyComponents/MyEvents';

let wrapper, instance, mounted;
let store, componentProps;

const getComponent = (props) => {
  mounted = shallow(<MyEvents {...props} />);
  return mounted;
};

describe('MyEvents component', () => {
  test('should be mounted when a user is authenticated', () => {
    componentProps = {
      activeState: {
        isAuthenticated: true
      }
    };
    wrapper = getComponent(componentProps);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.state('isAuthenticated')).toBeTruthy();
  });

  test('shoukld expect unauthorized user', () => {
    componentProps = {
      activeState: {
        isAuthenticated: false
      }
    };
    wrapper = getComponent(componentProps);
    expect(wrapper.state('isAuthenticated')).toBeFalsy();
  });
});
