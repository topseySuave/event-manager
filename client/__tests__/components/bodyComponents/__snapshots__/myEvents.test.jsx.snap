// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyEvents component MyEvents component should mount 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MyEvents
    activeState={
      Object {
        "isAuthenticated": true,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <div
          className="body__holdr"
        >
          <Connect(FixedNav) />
          <Connect(MyEventCardHolder) />
          <Connect(FloatingActionButton) />
          <Footer />
        </div>,
        <Modals />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Connect(FixedNav) />,
            <Connect(MyEventCardHolder) />,
            <Connect(FloatingActionButton) />,
            <Footer />,
          ],
          "className": "body__holdr",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <div
            className="body__holdr"
          >
            <Connect(FixedNav) />
            <Connect(MyEventCardHolder) />
            <Connect(FloatingActionButton) />
            <Footer />
          </div>,
          <Modals />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Connect(FixedNav) />,
              <Connect(MyEventCardHolder) />,
              <Connect(FloatingActionButton) />,
              <Footer />,
            ],
            "className": "body__holdr",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
